lib/realtime_web/controllers/tenant_controller.ex

```diff
@@ -26,6 +25,8 @@ defmodule RealtimeWeb.TenantController do
 
   action_fallback(RealtimeWeb.FallbackController)
 
+  plug :set_observability_attributes when action in [:show, :edit, :update, :delete, :reload, :health]
```

Comment:
@Ziinc I ended up using a `plug` here as we don't needed for all actions. I thought about moving this to the router but it would essentially need to look for `tenant_id` as path_params on all the routes which I don't know if that's what we want 🤷 

This felt like a good compromise?

Reply:
the plug can handle the situation where there isn't a tenant_id in path param with a  fallthrough so you don't have to explicitly state the actions, that would make it less verbose 
lgtm